src/examples/zkprogram/mututal-recursion.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  ZkProgram,
+  Field,
+  DynamicProof,
+  Proof,
+  VerificationKey,
+  Undefined,
+  verify,
+} from 'o1js';
+
+/**
+ * This example showcases mutual recursion (A -> B -> A) through two circuits that respectively
+ * add or multiply a given publicInput.
+ * Every multiplication or addition step consumes a previous proof from the other circuit to verify prior state. 
+ */
+
+class DynamicMultiplyProof extends DynamicProof<Undefined, Field> {
+  static publicInputType = Undefined;
+  static publicOutputType = Field;
+  static maxProofsVerified = 1 as const;
+}
+
+const add = ZkProgram({
```

Comment:
this example makes me a bit uncomfortable, since `add` isn't proving that `DynamicMultiplyProof` came from the multiply program, and that might not be obvious to users seeing this example.

I think adding a TODO comment where you highlight this caveat would be enough to prevent this pattern accidentally ending up in a user app and creating a vulnerability